src/main/java/io/confluent/connect/hdfs/tools/SchemaSourceTask.java

```diff
@@ -127,6 +158,89 @@ public void start(Map<String, String> props) {
 
   @Override
   public List<SourceRecord> poll() throws InterruptedException {
+    if ("logical".equalsIgnoreCase(this.schemaType)) {
+      return pollLogical();
+    } else if ("primitive".equalsIgnoreCase(this.schemaType)) {
+      return pollPrimitive();
+    } else {
+      throw new InterruptedException("Invalid schema type");
+    }
+  }
+
+  private List<SourceRecord> pollLogical() throws InterruptedException {
```

Comment:
This actually could extend the schema that is created by poll primitive. But even without that, I wonder if those methods could share some code.